tests/aws_iot_devicedefender_client_test.c

```diff
@@ -35,7 +40,18 @@
 #    include <unistd.h>
 #endif
 
-struct aws_iotdevice_defender_v1_task *defender_task = NULL;
+struct aws_iotdevice_defender_task *defender_task = NULL;
+struct aws_mutex stop_mutex = AWS_MUTEX_INIT;
+struct aws_condition_variable failure_stop_cv = AWS_CONDITION_VARIABLE_INIT;
+struct aws_condition_variable *process_stop_cv;
+
+void sigint_handler(int signal) {
+    (void)signal;
+    if (defender_task) {
+        /* not entirely sure if these are reentrant */
+        aws_condition_variable_notify_one(process_stop_cv);
```

Comment:
`notify_one` will notify the first waiter. If there's only ever 1, that's fine. If there are multiple, you probably want `notify_all`

Reply:
Removed the handler entirely